clients/src/main/java/org/apache/kafka/server/policy/AlterConfigPolicy.java

```diff
@@ -71,7 +71,7 @@ public int hashCode() {
 
         @Override
         public boolean equals(Object o) {
-            if (o == null || o.getClass() != o.getClass()) return false;
+            if ((o == null) || (!o.getClass().equals(getClass()))) return false;
```

Comment:
As far as I know, `Class` does not override `equals` so the behavior is similar to the previous one. Am I missing something?

Reply:
Disregard that, I read your comment wrong. The previous behavior was comparing the incoming object against itself. So not the same.

Reply:
Oh.. I missed that. It thought it was only about using `equals`. Could we keep using `!=` instead of `equals` though? It might be worth adding a small unit test for this method, what do you think? That would avoid repeating such mistake.

Reply:
`clients/src/main/java/org/apache/kafka/common/protocol/types/RawTaggedField.java` is doing the same thing fwiw. I don't have a strong preference other than I don't have the code checked out and it takes O(hours) to clone the repo. If you're fine either way or want to take this fix yourself, I don't mind.

Reply:
I don't feel strong about the `!=` vs. `equals`. We can keep it as you did it. However, I just took the liberty to push a small unit test to your branch. I hope you don't mind.

Reply:
Not at all, thanks!